### DISCUSSION TITLE
S3 selectObjectContent CSV output serialization not including headers

### DISCUSSION THREAD

**Discussion by travisemichael** [aws/aws-sdk-java-v2#3351] [Q&A]
I am using S3AsyncClient#selectObjectContent to query files within a bucket. When querying CSV files within the bucket that have a header line and setting the FileHeaderInfo to "USE" the records that are streamed back using CSV output serialization do not include the header line. Is there a way to include the header line when serializing the returned records as CSV?

**Comment by debora-ito** [comment]
By the documentation on the API reference it should be including the file header info.
Can you share a sample code we can use to reproduce the issue?
